src/NUnitTestAdapter/Internal/AsyncMethodHelper.cs

```diff
@@ -24,7 +24,7 @@ public string GetClassNameForAsyncMethod(string className, string methodName)
             var definingType = targetAssembly.GetType(className);
             if (definingType == null) return null;
 
-            var method = definingType.GetMethod(methodName);
+            var method = definingType.GetMethods().Where(o => o.Name == methodName).OrderBy(o => o.GetParameters().Length).FirstOrDefault();
```

Comment:
Actually, I'd like to know this for sure because I'm tired of avoiding Linq where it would be helpful. :-)